### ISSUE TITLE
TableEditor menu actions with on_perform are performed twice

### ISSUE THREAD

**Issue by k2bd** [enthought/traitsui#741]
Running the following script, right clicking the table, and pressing "Click me!" prints:
```
Called with 1 args!
Called with 0 args!
```

I believe the one-arg perform comes from the BaseTableEditor's `_perform` method, and the zero-arg perform comes from PyFace's Action's `perform` method.

https://github.com/enthought/traitsui/blob/48fc57f504b284d5c420b16b4330356550b448f3/traitsui/editors/table_editor.py#L520-L522

https://github.com/enthought/pyface/blob/03cbcfbe6e36167576a97462fa86ea9ef1b0ea79/pyface/action/action.py#L126-L135

```python
import random

from traits.api import HasTraits, List, Int, Instance
from traitsui.api import ModelView, TableEditor, Menu, Action, ObjectColumn, View, Item


class Result(HasTraits):
    value = Int

    def _value_default(self):
        return random.randint(1, 100)


class MyModel(HasTraits):
    results = List(Instance(Result))

    def _results_default(self):
        return [Result() for _ in range(20)]


def my_table(view):
    return TableEditor(
        menu=Menu(
            Action(
                name="Click me!",
                on_perform=view.do_perform,
            )
        ),
        columns=[
            ObjectColumn(
                name="value",
                label="A number",
                editable=False,
            ),
        ],
    )


class MyView(ModelView):
    model = Instance(MyModel)

    def default_traits_view(self):
        return View(
            Item(
                "model.results",
                editor=my_table(self),
            )
        )

    def do_perform(self, *args):
        print("Called with {} args!".format(len(args)))


if __name__ == "__main__":
    model = MyModel()
    view = MyView(model=model)

    view.configure_traits()
```

**Comment by k2bd** [comment]
As a follow-up, I see the same behaviour using a `Menu` on a `TabularAdapter`, the one-argument call coming from `qt4.editor.Editor._perform` and the zero-argument call coming from `pyface.action.action.Action.perform`.

**Comment by xamcost** [comment]
I'm reopening this issue as I encountered it, and I can actually still observe this behaviour with the code snippet given here.

**Comment by xamcost** [comment]
Reclosing, I was actually running on version 7.0.1, but that problem is solved on 7.1. I apologise for the noise !
